### PR TITLE
Buff ethanol and serotrotium

### PR DESCRIPTION
**Pull request by Kegdo**
## About The Pull Request

Alchohols give scaling sanity gain based on their strength
serotrotium insight gain = 3, previously - 1.5
## Why It's Good For The Game
Ciggaretes shall no longer be the king, long live throwing my wealth at alchohol at club.
## Changelog
:cl:
balance: Alcholic reagents give in general much more sanity based off their strength
balance: serotrotium insight gain increased to 3
/:cl:

### CODE REVIEW

**Comment by MLGTASTICa** [comment]
No meth , worst PR. 0/ 10

**Comment by Kegdo** [comment]
Then make the methe properly.

File: code/modules/sanity/sanity_mob.dm [hunk 1]
```diff
@@ -310,6 +310,9 @@
 	var/sanity_gain = E.sanity_gain_ingest
 	if(E.id == "ethanol")
 		sanity_gain /= 5
+	else if(istype(E, /datum/reagent/ethanol))
+		var/datum/reagent/ethanol/fine_drink = E
+		sanity_gain *= fine_drink.strength / 5
```
Comment (TheShown911) [inline code/modules/sanity/sanity_mob.dm:315]: Drink strength is reversed, lower is stronger, beer has 35 while vodka has 5.

Comment (Kegdo) [reply code/modules/sanity/sanity_mob.dm]: fixed, now it has a better formula (5 - X/5, X <= 20)

**Comment by MLGTASTICa** [comment]
Revert it back to being strength / 5 . Reason for this is that cocktails have it very high , its meant to scale off tastiness and work , not alchohol strength , these changes would serve only to make vodka the best drink , with working for other , more refined ones worthless.

**Comment by Humonitarian** [comment]
Cooked protein and nutriment is the biggest troll.
You can grind it down and drink it to restore sanity.
Just feels like shitcode in general, with all the hardcode and similar problems.

Scaling ethanol off of strength is a good idea, but it scales exponentially.

Strength 20 has a multiplier of 1, while strength 10 has a multiplier of 3. I feel it should do the opposite, diminishing bonuses with strength to keep weaker alcohols useful.

Also, is the taste_mult that drinks have taken into account anywhere?

**Comment by Humonitarian** [comment]
> Revert it back to being strength / 5 . Reason for this is that cocktails have it very high , its meant to scale off tastiness and work , not alchohol strength , these changes would serve only to make vodka the best drink , with working for other , more refined ones worthless.

Taste_mult. Please read the code before changing it.

**Comment by Kegdo** [comment]
> Cooked protein and nutriment is the biggest troll. You can grind it down and drink it to restore sanity.

I know, intended.

> Strength 20 has a multiplier of 1, while strength 10 has a multiplier of 3. I feel it should do the opposite, diminishing bonuses with strength to keep weaker alcohols useful.

15 has a multiplier of 3, but whatever; I think it's the most optimal solution, while also not making weak booze(like sake, 25 str) as powerful as strong ones, while still making the latter good.

> Also, is the taste_mult that drinks have taken into account anywhere?

who?

**Comment by Humonitarian** [comment]
> 15 has a multiplier of 3, but whatever; I think it's the most optimal solution, while also not making weak booze(like sake, 25 str) as powerful as strong ones, while still making the latter good.

sanity_gain *= (5 - (fine_drink.strength / 5))

When fine_drink == 20
sanity_gain *= (5 - (20 / 5))
= 5 - 4
= 1

When fine_drink == 10
sanity_gain *= (5 - (10 / 5))
= 5 - 2
= 3

When fine_drink == 15
sanity_gain *= (5 - (15 / 5))
= 5 - 3
= 2

15 is twice as good as 20, 10 is thrice as good.
If you want to solve something by math, test it. Type it in a calculator. Don't make me have to write down equations for you, please. You're not in elementary school. (You're not, right? Right?!)

> Who?

A multiplier used for cocktails to make them superior to other alcohol when fulfilling rest and insight. Literally exactly what you're looking for.

**Comment by Kegdo** [comment]
> 15 is twice as good as 20, 10 is thrice as good. You're not in elementary school. (You're not, right? Right?!)

bro I meant the sanity gain thingy, not that it's twice/thrice as good.............................................................................
yes I am in elementary school

> > Who?
> 
> A multiplier used for cocktails to make them superior to other alcohol when fulfilling rest and insight. Literally exactly what you're looking for.

Who?

**Comment by Kegdo** [comment]
taste_mult doesn't do anything. Only affects how noticeable the taste is.

**Comment by Humonitarian** [comment]
> bro I meant the sanity gain thingy, not that it's twice/thrice as good.............................................................................
yes I am in elementary school

Then my request makes sense now?

> taste_mult doesn't do anything. Only affects how noticeable the taste is.

Welp. Was worth a shot to look at.

**Comment by MLGTASTICa** [comment]
> > bro I meant the sanity gain thingy, not that it's twice/thrice as good.............................................................................
> > yes I am in elementary school
> 
> Then my request makes sense now?
> 
> > taste_mult doesn't do anything. Only affects how noticeable the taste is.
> 
> Welp. Was worth a shot to look at.

thats why ive been using strength at first , its very high for cocktails and generally low for raw drinks , which makes it perfect.

**Comment by Kegdo** [comment]
*beer(35), gargle blaster(1) and atom vodka would like to know your location*

**Comment by Humonitarian** [comment]
> _beer(35), gargle blaster(1) and atom vodka would like to know your location_

This. Don't cuck the gargleblaster.

My suggestion is to add a variable to Ethanol as "insight multiplier", and leave it at 1 for now, while rewriting the calculation to be balanced to alcohol content, with the multiplier included.

Something along the lines of 
sanity_gain *= (50 / (fine_drink.strength + 10))
sanity_gain *= insight_mult

**Comment by TGCMbetcher** [comment]
>seritronium
>Very hard to make
Im sorry what? Thanks for the buff but NT has the stuff to make it round start.

**Comment by Kegdo** [comment]
Sure they can, uh-huh. They still need to grow the musroom to get psilocybin. And it will take a lot of time. Moebius, the ones that actually make chems, can't even grow plants.

**Comment by Kegdo** [comment]
> Something along the lines of 
> sanity_gain *= (50 / (fine_drink.strength + 10))
> sanity_gain *= insight_mult

Sounds like something that will not be used, or will have to be adjusted for every cocktail there is

**Comment by Humonitarian** [comment]
> > Something along the lines of
> > sanity_gain *= (50 / (fine_drink.strength + 10))
> > sanity_gain *= insight_mult
> 
> Sounds like something that will not be used, or will have to be adjusted for every cocktail there is

Strength is adjusted for every cocktail there is

I mean, the alternative is a code that recognises the complexity of the cocktail based on recipe.

Eitherways, cocktails made from non-alcohols are usually more efficient insight-wise as they require less alcohol to make / unit.

**Review by infraredbaron — APPROVED** [review]
Code seems alright to me. Changes are really good as well

**Review by drexample — CHANGES_REQUESTED** [review]
Cooked nutriment/protein make absolutely no sense and is one of the worst ways to handle sanity gain. Players would be able to grind food and extract the specific cooked protein, making reagent mush extracted from blended food much more sanity restoring than the food itself.

**Comment by Kegdo** [comment]
Okie, removed the food buff.

**Comment by Humonitarian** [comment]
Still pressing my suggestion of using this formula, or something similar to it:

sanity_gain *= (50 / (fine_drink.strength + 10))

**Review by drexample — APPROVED** [review]

**Pull request merged**